### PR TITLE
Make error handling/token refresh better in trees latest

### PR DESCRIPTION
**Pull request by jatgarg**
I don't think there is need to have token refresh code so up the ladder in trees latest call. So moved it to the lowest level.

### CODE REVIEW

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 1]
```diff
@@ -658,13 +628,17 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
...
                     this.odspResolvedUrl,
                     this.getStorageToken,
-                    tokenFetchOptions,
                     snapshotOptionsWithoutBlobs,
                     this.logger,
                     snapshotDownloader,
                     putInCache);
                 return odspSnapshot;
             }
+            // Clear the cache on 401/403/404 on snapshot fetch from network because this means either the user doesn't have
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:637]: I think logically this should move to fetchLatestSnapshotCore(), as it is applicable to pre-fetching as well, right?

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 2]
```diff
@@ -658,13 +628,17 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
...
+            // permissions for the file or it was deleted. So the user will again try to fetch from cache on any failure in future.
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:638]: maybe clarify it:
"So, if we do not clear cache, we will continue fetching snapshot from cache in the future."

**Comment by vladsud** [comment]
Unrelated to this change, but I'd remove prefetch property from this event - naming prefix is more than enough to differentiate, we should use either name or property, but not both. I like prefix better - it's easier to work with data that way.

>    const odspLogger = createOdspLogger(ChildLogger.create(logger, "PrefetchSnapshot", { all: { prefetch: true }}));

**Review by vladsud — APPROVED** [review]

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts (hunk 1 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:637]: I will handle this in my next PR about redeeming.

**Pull request merged**